engine/src/main/java/org/terasology/network/internal/NetworkSystemImpl.java

```diff
@@ -784,6 +785,10 @@ void removeClient(NetClient client) {
 
     private void processRemovedClient(Client client) {
         if (client instanceof NetClient) {
+            ServerConnectListManager serverConnectListManager = context.get(ServerConnectListManager.class);
```

Comment:
Is the purpose of this part to avoid doing the cleanup performed by `processRemovedClient` in case the client was immediately disconnected due to not being allowed, and thus was never actually fully connected (i.e. not added to `clientList`, `netClientList`, `clientPlayerLookup`, etc)?

(not implying I disagree with anything, just asking because I don't have much experience with the NetworkSystem and I'm curios whether I'm understanding it correctly)

Reply:
Good, thanks for the explanation